src/main/java/overlap_use_case/TestTimetableModelUnpacker.java

```diff
@@ -0,0 +1,101 @@
+package overlap_use_case;
+
+import entities.*;
+import org.junit.Assert;
+import org.junit.Test;
+import retrieve_timetable_use_case.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class TestTimetableModelUnpacker {
+
+    private BlockModel basicBlockModel = new BlockModel(1, 18.00, 20.00, "Castle Badr");
+    private Block basicBlock = new Block("TU", "18:00", "20:00", "Castle Badr");
+
+    private SectionModel basicSectionModel = new SectionModel("TES101", "Mario-chan", List.of(basicBlockModel));
+    private Section basicSection = new Section("TES101", "Mario-chan", List.of(basicBlock));
+
+    private CourseModel basicCourseModel = new CourseModel("TES101", List.of(basicSectionModel), "S",
+            "TES101", "1");
+
+    private CalendarCourse basicCalendarCourse = new CalendarCourse("TES101", List.of(basicSection), "S",
+            "TES101", "1");
+
+    private TimetableCourse basicTimetableCourse = new TimetableCourse("TES101", List.of(basicSection), "S",
+            "TES101", "1");
+
+    public TestTimetableModelUnpacker() throws InvalidSectionsException {
+    }
+
+
+    /** Test unpackBlockModel with a basic case. **/
+    @Test
+    public void test_unpack_block_model_basic(){
```

Comment:
I would rename these methods so that they fall in line with our camelCaseMethodName conventions. Overall test coverage here looks good!

Reply:
Done!